src/brogue/Items.c

```diff
@@ -6032,6 +6047,44 @@ void relabel(item *theItem) {
     }
 }
 
+// If the most recently equipped item caused another item to be unequiped and
+// both haven't left the inventory since, swap them back.
+void swap() {
+    item *theItem;
+    char buf[COLS * 3], swappedOutName[COLS], swappedInName[COLS];
+    unsigned char command[10];
+
+    if (rogue.swappedIn == NULL || rogue.swappedOut == NULL) {
+        confirmMessages();
+        message("You have nothing to swap.", false);
+        return;
+    }
+
+    command[0] = SWAP_KEY;
+    command[1] = '\0';
+    recordKeystrokeSequence(command);
+
+    unequipItem(rogue.swappedIn, false);
+    if (rogue.swappedIn->category & RING) {
+        updateRingBonuses();
+    }
+
+    equipItem(rogue.swappedOut, false);
+    strengthCheck(rogue.swappedOut);
+
+    theItem = rogue.swappedIn;
+    rogue.swappedIn = rogue.swappedOut;
+    rogue.swappedOut = theItem;
+
+    confirmMessages();
+    itemName(rogue.swappedOut, swappedOutName, true, false, NULL);
+    itemName(rogue.swappedIn, swappedInName, true, true, NULL);
+    sprintf(buf, "Now %s %s instead of your %s.", (rogue.swappedIn->category & WEAPON ? "wielding" : "wearing"), swappedInName, swappedOutName);
```

Comment:
If the currently equipped gear is cursed, `unequipItem` and `equipItem` both do the right thing.

However this messaging (and the pointer swap above) are bugged.

Reply:
Fixed